nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/util/TextLineDemarcator.java

```diff
@@ -95,52 +71,61 @@ public OffsetInfo nextOffsetInfo() {
      *
      * @return offset info
      */
-    public OffsetInfo nextOffsetInfo(byte[] startsWith) {
+    public OffsetInfo nextOffsetInfo(byte[] startsWith) throws IOException {
```

Comment:
I'm not sure that i understand the concept here of passing in a `byte[] startsWith`. The method returns an OffsetInfo regardless. Whether or not it starts with some arbitrary byte array seems like a very separate concern. Seems like that should be externalized and let the caller check if it starts with some pattern... or offering a `startsWith(byte[])` method on OffsetInfo?